packages/typegpu/src/core/buffer/buffer.ts

```diff
@@ -58,6 +62,12 @@ export function INTERNAL_createBuffer<TData extends AnyData>(
   typeSchema: TData,
   initialOrBuffer?: Infer<TData> | GPUBuffer,
 ): TgpuBuffer<TData> {
+  if (isLooseArray(typeSchema) || isLooseStruct(typeSchema)) {
```

Comment:
Tiny nit, can be made more general in case we add more loose schemas.
```suggestion
  if (!isWgslData(typeSchema)) {
```